Build failure: zeroad

Another victim of the recent libxml2 changes on master. @chvp 

### Build log

```
../../../source/ps/XML/RelaxNG.cpp: In member function 'bool RelaxNGValidator::ValidateEncoded(xmlDocPtr) const':
../../../source/ps/XML/RelaxNG.cpp:162:50: error: invalid conversion from 'void (*)(void*, xmlErrorPtr)' {aka 'void (*)(void*, _xmlError*)'} to 'xmlStructuredErrorFunc' {aka 'void (*)(void*, const _xmlError*)'} [-fpermissive]
  162 |         xmlRelaxNGSetValidStructuredErrors(ctxt, &relaxNGErrorHandler, NULL);
      |                                                  ^~~~~~~~~~~~~~~~~~~~
      |                                                  |
      |                                                  void (*)(void*, xmlErrorPtr) {aka void (*)(void*, _xmlError*)}
In file included from ../../../source/ps/XML/RelaxNG.cpp:28:
/nix/store/iqjsxkcdnvvz1bfpq960ygicc5clz9hv-libxml2-2.12.3-unstable-2023-12-14-dev/include/libxml2/libxml/relaxng.h:185:66: note:   initializing argument 2 of 'void xmlRelaxNGSetValidStructuredErrors(xmlRelaxNGValidCtxtPtr, xmlStructuredErrorFunc, void*)'
  185 |                                           xmlStructuredErrorFunc serror, void *ctx);
      |                                           ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
Xeromyces.cpp
../../../source/ps/XML/Xeromyces.cpp: In static member function 'static void CXeromyces::Startup()':
../../../source/ps/XML/Xeromyces.cpp:59:41: error: invalid conversion from 'void (*)(void*, xmlErrorPtr)' {aka 'void (*)(void*, _xmlError*)'} to 'xmlStructuredErrorFunc' {aka 'void (*)(void*, const _xmlError*)'} [-fpermissive]
   59 |         xmlSetStructuredErrorFunc(NULL, &errorHandler);
      |                                         ^~~~~~~~~~~~~
      |                                         |
      |                                         void (*)(void*, xmlErrorPtr) {aka void (*)(void*, _xmlError*)}
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/games/0ad/game.nix
+++ b/pkgs/games/0ad/game.nix
@@ -55,7 +55,24 @@ stdenv.mkDerivation rec {
     "-L${nvidia-texture-tools.lib}/lib/static"
   ];
 
-  patches = [ ./rootdir_env.patch ];
+  patches = [
+    ./rootdir_env.patch
+
+    # Fix build with libxml v2.12
+    # FIXME: Remove with next package update
+    (fetchpatch {
+      name = "libxml-2.12-fix.patch";
+      url = "https://github.com/0ad/0ad/commit/d242631245edb66816ef9960bdb2c61b68e56cec.patch";
+      hash = "sha256-Ik8ThkewB7wyTPTI7Y6k88SqpWUulXK698tevfSBr6I=";
+    })
+    # Fix build with GCC 13
+    # FIXME: Remove with next package update
+    (fetchpatch {
+      name = "gcc-13-fix.patch";
+      url = "https://github.com/0ad/0ad/commit/093e1eb23519ab4a4633a999a555a58e4fd5343e.patch";
+      hash = "sha256-NuWO64narU1JID/F3cj7lJKjo96XR7gSW0w8I3/hhuw=";
+    })
+  ];
 
   configurePhase = ''
     # Delete shipped libraries which we don't need.

